components/RegionDialog/RegionDialog.tsx

```diff
@@ -0,0 +1,104 @@
+import { useRouter } from "next/router";
+import React from "react";
+import { useForm } from "react-hook-form";
+
+import { clearCheckout } from "@/lib/checkout";
+import { DEFAULT_LOCALE, LOCALES } from "@/lib/regions";
+
+import { Button } from "../Button";
+import useChannels from "../ChannelsProvider/useChannels";
+
+interface RegionDialogProps {
+  onClose: () => void;
+  isOpen: boolean;
+}
+
+export interface RegionFormData {
+  channel: string;
+  locale: string;
+}
+
+const RegionDialog = ({ isOpen, onClose }: RegionDialogProps) => {
+  const router = useRouter();
+  const { channels, currentChannel, setCurrentChannel } = useChannels();
+
+  const { register, handleSubmit } = useForm<RegionFormData>({
+    defaultValues: {
+      channel: currentChannel.slug,
+      locale: router.locale || DEFAULT_LOCALE,
+    },
+  });
+
+  const onSubmit = handleSubmit(async (formData: RegionFormData) => {
+    if (formData.channel !== currentChannel.slug) {
+      await setCurrentChannel(formData.channel);
+      clearCheckout();
+    }
+    onClose();
+
+    // Update current URL to use the chosen channel
+    await router.push({
```

Comment:
`await` not needed

Reply:
Had strange behavior without it, please test the change. Not sure if it was a coincidence or a real issue.